src/utils/vacation-data-utils.tsx

```diff
@@ -0,0 +1,42 @@
+import { DailyEntry } from "generated/client";
+import { VacationDayData, VacationWeekData } from "types";
+import getISOWeek from "date-fns/getISOWeek";
+
+/**
+ * Utility class for vacation days
+ */
+export default class VacationDataUtils {
+
+  /**
+   * Preprocess list for vacation days and weeks
+   * 
+   * @param dailyEntry daily entries from api request
+   * @return weeks and days of vacation 
+   */
+  public static vacationDaysProcess = (dailyEntries: DailyEntry[]): VacationWeekData[] => {
+    const vacationDays: VacationDayData[] = dailyEntries
+      .filter(dailyEntry => dailyEntry.isVacation)
+      .map(dailyEntry => {
+        return {
```

Comment:
Remove unnecessary return.

Reply:
Is it unnecessary? See that the array is of type DailyEntry and we want to return it as VacationDayData.